src/setting.rs

```diff
@@ -44,65 +58,24 @@ pub struct Setting {
     /// details.
     pub checksum: u16,
     /// Unknown field (apparently always `0000`).
+    #[br(assert(unknown == 0))]
```

Comment:
```suggestion
    #[brw(assert(unknown == 0))]
```

Reply:
This prevents serializing other values though. Do we really want that?

Reply:
Well, do we have a reason to serialize other values right now? IMO we can still change that and lifting that restriction is not a breaking change.

Reply:
> Well, do we have a reason to serialize other values right now?

In theory, a user could fill in the values by hand and then serialize it.

> IMO we can still change that and lifting that restriction is not a breaking change.

Sure. Probably makes sense to add it.

Unfortunately I can't get it to work. Sometimes `unknown` is an `u16`, sometimes it's apparently an `&u16`?!

```rust
error[E0277]: can't compare `&u16` with `{integer}`
  --> src/setting.rs:61:26
   |
61 |     #[brw(assert(unknown == 0))]
   |                          ^^ no implementation for `&u16 == {integer}`
   |
```

```rust
error[E0277]: can't compare `&u16` with `u16`
  --> src/setting.rs:61:26
   |
61 |     #[brw(assert(unknown == 0u16))]
   |                          ^^ no implementation for `&u16 == u16`
   |
```

```rust
error[E0308]: mismatched types
  --> src/setting.rs:61:29
   |
61 |     #[brw(assert(unknown == &0u16))]
   |                             ^^^^^ expected `u16`, found `&u16`
   |
```

```rust
error[E0614]: type `u16` cannot be dereferenced
  --> src/setting.rs:61:18
   |
61 |     #[brw(assert(*unknown == 0u16))]
   |                  ^^^^^^^^
```

If you want to have a look, feel free to do so :D

Reply:
Mhmm, I think thats a bug. I feel like `brw` just gets expanded into a `br` and `bw` macro which have different types for their asserts then or something.